client/layout/guided-tours/index.js

```diff
@@ -22,24 +23,52 @@ class GuidedTours extends Component {
 		return this.props.tourState !== nextProps.tourState;
 	}
 
-	next = ( tour, nextStepName ) => {
+	start = ( { tour, tourVersion: tour_version } ) => {
+		if ( tour && tour_version ) {
+			tracks.recordEvent( 'calypso_guided_tours_start', {
+				tour,
+				tour_version,
+			} );
+		}
+	}
+
+	next = ( { step, tour, tourVersion, nextStepName, skipping = false } ) => {
+		if ( ! skipping && step ) {
+			tracks.recordEvent( 'calypso_guided_tours_seen_step', {
+				tour,
+				step,
+				tour_version: tourVersion,
+			} );
+		}
+
 		defer( () => {
 			this.props.nextGuidedTourStep( {
+				tour,
 				stepName: nextStepName,
-				tour: tour,
 			} );
 		} );
 	}
 
-	quit = ( options = {} ) => {
-		this.props.quitGuidedTour( Object.assign( {
-			stepName: this.props.tourState.stepName,
-			tour: this.props.tourState.tour,
-		}, options ) );
-	}
+	quit = ( { step, tour, tourVersion: tour_version, isLastStep } ) => {
+		if ( step ) {
+			tracks.recordEvent( 'calypso_guided_tours_seen_step', {
```

Comment:
This one is AFAIK redundant, as a `quit` or `finished` event is fired with the same data immediately after.

Whether we want this last `_seen_step` action or not is a matter of semantics. For me, it makes sense to keep.